test/runtime/lb.go

```diff
@@ -71,231 +70,249 @@ var _ = Describe("RuntimeValidatedLB", func() {
+				fmt.Sprintf("%s:80", httpd2[helpers.IPv4]),
+			},
+			"[f00d::1:1]:80": {
+				fmt.Sprintf("[%s]:80", httpd1[helpers.IPv6]),
+				fmt.Sprintf("[%s]:80", httpd2[helpers.IPv6]),
+			},
+		}
+		svc := 1
+		for fe, be := range services {
+			status := vm.ServiceAdd(svc, fe, be)
+			status.ExpectSuccess("failed to create service %s=>%v, fe, be")
+			svc++
+		}
 
-		By("L3 redirect to L4")
-		status = vm.ServiceAdd(3, "2.2.2.2:0", []string{
+		By("Making HTTP requests from container => bpf_lb => container")
 
-			fmt.Sprintf("%s:80", httpd1[helpers.IPv4]),
-			fmt.Sprintf("%s:80", httpd2[helpers.IPv4])}, 2)
-		status.ExpectFail("Service created with invalid data")
+		for ip := range services {
+			url := fmt.Sprintf("http://%s/public", ip)
+			status := vm.ContainerExec(helpers.Client, helpers.CurlFail(url))
+			status.ExpectSuccess(fmt.Sprintf("failed to fetch via URL %s", url))
+		}
 	}, 500)
 
-	It("Service recovery on restart", func() {
+	It("validates service recovery on restart", func() {
 		service := "2.2.2.2:80"
 		svcID := 1
-		testCmd := fmt.Sprintf(
-			"curl -s --fail --connect-timeout 4 http://%s/public", service)
-
-		containers("create")
-		epStatus := vm.WaitEndpointsReady()
-		Expect(epStatus).Should(BeTrue())
+		testCmd := helpers.CurlFail(fmt.Sprintf("http://%s/public", service))
 
+		createContainers()
 		httpd1, err := vm.ContainerInspectNet("httpd1")
 		Expect(err).Should(BeNil())
-
 		httpd2, err := vm.ContainerInspectNet("httpd2")
 		Expect(err).Should(BeNil())
 
 		status := vm.ServiceAdd(svcID, service, []string{
 			fmt.Sprintf("%s:80", httpd1["IPv4"]),
-			fmt.Sprintf("%s:80", httpd2["IPv4"])}, 2)
-		status.ExpectSuccess("L4 service can't be created")
+			fmt.Sprintf("%s:80", httpd2["IPv4"])})
+		status.ExpectSuccess("failed to create service %s=>{httpd1,httpd2}", service)
 
-		status = vm.ContainerExec("client", testCmd)
-		status.ExpectSuccess("L4 Proxy is not working IPv4")
+		By("Making HTTP request via the service before restart")
 
+		status = vm.ContainerExec(helpers.Client, testCmd)
+		status.ExpectSuccess("Failed to fetch URL via service")
 		oldSvc := vm.ServiceList()
 		oldSvc.ExpectSuccess("Cannot retrieve service list")
 
+		By("Fetching service state before restart")
+
 		oldSvcIds, err := vm.ServiceGetIds()
 		Expect(err).Should(BeNil())
 		oldBpfLB, err := vm.BpfLBList()
 		Expect(err).Should(BeNil())
 
-		res := vm.Exec("sudo systemctl restart cilium")
-		res.ExpectSuccess()
-
-		err = vm.WaitUntilReady(100)
+		err = vm.RestartCilium()
 		Expect(err).Should(BeNil())
 
-		epStatus = vm.WaitEndpointsReady()
-		Expect(epStatus).Should(BeTrue())
+		By("Checking that the service was restored correctly")
 
 		svcIds, err := vm.ServiceGetIds()
 		Expect(err).Should(BeNil())
-		Expect(len(svcIds)).Should(Equal(len(oldSvcIds)), "Service recovery does not match")
-
+		Expect(len(svcIds)).Should(Equal(len(oldSvcIds)),
+			fmt.Sprintf("Service ids %s do not match old service ids %s", svcIds, oldSvcIds))
```

Comment:
You don't need to have the fmt.Sprintf here ;-)

Reply:
Ah. I get caught by this kind of thing all the time, so I instinctively wrap things in `fmt.Sprintf()` ;) ...I can remove this.